apps/publish/content/common.py

```diff
@@ -895,8 +895,7 @@ def _publish_associated_items(self, original, updates=None):
                         sync_associated_item_changes(associated_item, associated_item_updates)
                         continue
 
-                    if association_updates.get("state") not in PUBLISH_STATES:
-                        # There's an update to the published associated item
+                    if association_updates.get("state") not in PUBLISH_STATES or updates.get("state"):
```

Comment:
I would rather compare the data, eg headline, slugline, etc. Imo this is again always true.
what we want to do is publish the association_updates if there are any changes, but not if there were no changes, updating that image metadata and republishing all images is kind of expensive